### PR TITLE
Updated find method.

### PR DESCRIPTION
**Pull request by kgirvan**

### CODE REVIEW

File: lib/ks/curriculum/curriculum.rb [hunk 1]
```diff
@@ -1549,16 +1549,20 @@ class Curriculum
...
-  def find(type, name, opts={})
+  def find(type, course_code, course_name, username,  opts={})
     
     defaults = {
       :nav_homepage => true,
       :course_description => '',
-      :course_number => ''
+      :course_number => '',
+      :course_name_dyn_var => 'course_name_dyn_var',
+      :course_name_var_regexp => '\"\([^\"]+\)\"\,\"[^\"]+\"\,\"'+course_name+'\"'
```
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: The course_name variable is not going to be evaluated as written, it's inside a single quote string. You are going to have to concatenate that string to get the variable to evaluate.

File: lib/ks/curriculum/curriculum.rb [hunk 2]
```diff
@@ -1569,10 +1573,212 @@ class Curriculum
...
+          'contents' => "5|0|13|#{@request.url}/org.kuali.student.lum.lu.ui.main.LUMMain/|648421FAE6C751B6B3D6A2EC5262F586|org.kuali.student.common.ui.client.service.SearchRpcService|search|org.kuali.student.core.search.dto.SearchRequest|org.kuali.student.core.search.dto.SearchRequest/3917446114|java.lang.Boolean/476441737|java.util.ArrayList/3821976829|org.kuali.student.core.search.dto.SearchParam/3876231949|proposal.queryParam.proposalOptionalName|#{course_code}|proposal.search.generic|proposal.resultColumn.proposalOptionalName|1|2|3|4|1|5|6|0|7|0|8|1|9|10|0|11|12|13|0|0|"
         }
       )
     elsif(type == "course")
+#Ken is this it
```
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: Remove comment

File: lib/ks/curriculum/curriculum.rb [hunk 3]
```diff
@@ -1569,10 +1573,212 @@ class Curriculum
...
+else
+  contents = 
+"5|0|20|#{@request.url}/org.kuali.student.lum.lu.ui.main.LUMMain/|DB85114A8D2B33860498043707FB831D|org.kuali.student.common.ui.client.service.SearchRpcService|search|org.kuali.student.common.search.dto.SearchRequest/2597477947|java.lang.Integer/3438268394|java.lang.Boolean/476441737|java.util.ArrayList/3821976829|org.kuali.student.common.search.dto.SearchParam/1222427352|lu.queryParam.luOptionalCode|#{course_code}|lu.queryParam.luOptionalType|kuali.lu.type.CreditCourse|lu.queryParam.luOptionalState|java.lang.String/2004016611|Approved|Active|Retired|lu.search.mostCurrent.union|lu.resultColumn.luOptionalCode|1|2|3|4|1|5|5|6|10|7|0|8|3|9|10|0|11|9|12|0|13|9|14|8|3|15|16|15|17|15|18|0|19|20|0|6|0|"
+end
```
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: Fix indentation of lines 1581-1588

File: lib/ks/curriculum/curriculum.rb [hunk 4]
```diff
@@ -1584,7 +1790,7 @@ class Curriculum
           'contents' => contents
         }
       )
-
+=end
```
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: I don't understand this begin block, why is it there?

File: lib/ks/curriculum/curriculum.rb [hunk 5]
```diff
@@ -1569,10 +1573,212 @@ class Curriculum
...
         }
       )
     elsif(type == "course")
+#Ken is this it
+# Only searching by title/name
+if(course_code.empty?)
+  contents = 
+"5|0|18|#{@request.url}/org.kuali.student.lum.lu.ui.main.LUMMain/|DB85114A8D2B33860498043707FB831D|org.kuali.student.common.ui.client.service.SearchRpcService|search|org.kuali.student.common.search.dto.SearchRequest/2597477947|java.lang.Integer/3438268394|java.lang.Boolean/476441737|java.util.ArrayList/3821976829|org.kuali.student.common.search.dto.SearchParam/1222427352|lu.queryParam.luOptionalType|kuali.lu.type.CreditCourse|lu.queryParam.luOptionalState|java.lang.String/2004016611|Approved|Active|Retired|lu.search.mostCurrent.union|lu.resultColumn.luOptionalCode|1|2|3|4|1|5|5|6|10|7|0|8|2|9|10|0|11|9|12|8|3|13|14|13|15|13|16|0|17|18|0|6|0|"
```
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: What happens if course_code is empty? Your if/else block only sets the contents variable but never sends any transactions through if course_code is empty.

File: tests/ks/course_search.rb [hunk 6]
```diff
@@ -37,8 +39,17 @@ auth.login({:user => username, :password => password})
 # Search for various courses
 cp_req = sesh.add_transaction("course_nav").add_requests
```
Comment (kcampos) [inline tests/ks/course_search.rb]: I'd rename cp_req variable, this is a cut and paste from create proposal, hence the "cp" abbreviation. Make this more obvious what it's for.

File: tests/ks/course_search.rb [hunk 7]
```diff
@@ -37,8 +39,17 @@ auth.login({:user => username, :password => password})
...
+course_code = "BSCI"
+transaction = "full_match"
+course_name = "Insects"
 curr_obj.homepage
 
+config.log.info_msg("#{test}: Search for course '#{course_code}'")
+req_obj = sesh.add_transaction(transaction).add_requests
+curr_obj.find('course', course_code, course_name, username, {:nav_homepage => false})
+
+=begin
```
Comment (kcampos) [inline tests/ks/course_search.rb]: again not sure I see the point of the begin block here

**Comment by kcampos** [comment]
See my inline comments on your code, please checkin fixes before I accept the pull request.

File: lib/ks/curriculum/curriculum.rb (hunk 5 above)
Comment (kgirvan) [inline lib/ks/curriculum/curriculum.rb]: transactions are sent in the following which is after the if else end block:
      @request.add('/org.kuali.student.lum.lu.ui.main.LUMMain/rpcservices/SearchRpcService',
        {
          'method' => 'POST',
          'content_type' => 'text/x-gwt-rpc; charset=utf-8',
          'contents' => contents
        },
        {
          :dyn_variables => [
            {"name" => opts[:course_name_dyn_var], "regexp" => opts[:course_name_var_regexp]}
          ]
        }
      )

File: lib/ks/curriculum/curriculum.rb (hunk 1 above)
Comment (kgirvan) [inline lib/ks/curriculum/curriculum.rb]: Kenneths-MacBook-Pro:Kuali-Sakai-Performance-Test-Automation-Framework girvank$ ./driver.rb
patchlevel: 290, release_date: 2011-07-09, ruby_version: 1.9.2, ruby_platform: x86_64-darwin11.1.0
Your log file is here: /Users/girvank/work/Kuali-Sakai-Performance-Test-Automation-Framework/log/1318949110.log
Let's setup your test run config...
What product are we testing? ["ks", "oae"] ks
What suite are we testing? search

What do you want to name your output xml file? ken1018b.xml
Your xml file will be here: /Users/girvank/work/Kuali-Sakai-Performance-Test-Automation-Framework/config/tests/ken1018b.xml
What clients are you driving your tests from? [hostname1,hostname2,...] localhost
What primary servers do you want to test? [hostname1:port,hostname2:port,...] ec2-72-44-53-140.compute-1.amazonaws.com:80
Secondary servers? [reference_name1:hostname1:port,reference_name2:hostname2:port,...] rice:ec2-50-16-13-113.compute-1.amazonaws.com:80
What is the primary server context? ks-standalone
What is the secondary server context? ks-rice
Let's setup your test scenario or phases, you can define multiple phases
Phase 1: How many minutes? 1
At what interval should users be created at this phase? (x seconds per user) 59
Do you want to add another phase? [y/n] n
Let's setup what user agents you'd like to test with
Here are the agents you can specify...
1: Mozilla/5.0 (Windows; MSIE 7.0; Windows NT 6.0; en-US)
2: Mozilla/4.0 (Windows; MSIE 6.0; Windows NT 6.0)
3: Mozilla/5.0 (Windows; U; Windows NT 6.0; en-US)
4: Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.6; en-US; rv:1.9.1.6) Gecko/20091201 Firefox/3.5.6
Specify the number of the agent you'd like to use: 1
Specify the probability that this agent will be used: [1-100] 100
Do you want to insert user thinktime? [y/n] y
Do you want to execute tsung now? [y/n] n
Do you want to dump http traffic from tsung? [y/n] y

Test: course_search.rb
Probability: 100
course_search.rb: Logging in as: fishdptcoc1/fishdptcoc1
course_search.rb: Search for course 'BSCI'
course_search.rb: Logging out
:course_name_var_regexp = \"([^\"]+)\"\,\"[^\"]+\"\,\"Insects\"  <-- output from a puts statement I added to curriculum.rb

Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: Oh I see you did concatenate it with the +, can you add spaces around the plus to make that more visible. Thx

File: lib/ks/curriculum/curriculum.rb (hunk 5 above)
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: OK please fix indentation, hard to read.

File: lib/ks/curriculum/curriculum.rb [hunk 8]
```diff
@@ -1569,22 +1572,207 @@ class Curriculum
...
-      
+      if(course_code.empty?)
+        contents = 
+"5|0|18|#{@request.url}/org.kuali.student.lum.lu.ui.main.LUMMain/|DB85114A8D2B33860498043707FB831D|org.kuali.student.common.ui.client.service.SearchRpcService|search|org.kuali.student.common.search.dto.SearchRequest/2597477947|java.lang.Integer/3438268394|java.lang.Boolean/476441737|java.util.ArrayList/3821976829|org.kuali.student.common.search.dto.SearchParam/1222427352|lu.queryParam.luOptionalType|kuali.lu.type.CreditCourse|lu.queryParam.luOptionalState|java.lang.String/2004016611|Approved|Active|Retired|lu.search.mostCurrent.union|lu.resultColumn.luOptionalCode|1|2|3|4|1|5|5|6|10|7|0|8|2|9|10|0|11|9|12|8|3|13|14|13|15|13|16|0|17|18|0|6|0|"
```
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb]: These content lines all the way down the file need proper indentation, it should be tabbed past "contents" column

File: lib/ks/curriculum/curriculum.rb [hunk 9]
```diff
@@ -1569,22 +1572,207 @@ class Curriculum
...
+      else
+        contents = 
+"5|0|20|#{@request.url}/org.kuali.student.lum.lu.ui.main.LUMMain/|DB85114A8D2B33860498043707FB831D|org.kuali.student.common.ui.client.service.SearchRpcService|search|org.kuali.student.common.search.dto.SearchRequest/2597477947|java.lang.Integer/3438268394|java.lang.Boolean/476441737|java.util.ArrayList/3821976829|org.kuali.student.common.search.dto.SearchParam/1222427352|lu.queryParam.luOptionalCode|#{course_code}|lu.queryParam.luOptionalType|kuali.lu.type.CreditCourse|lu.queryParam.luOptionalState|java.lang.String/2004016611|Approved|Active|Retired|lu.search.mostCurrent.union|lu.resultColumn.luOptionalCode|1|2|3|4|1|5|5|6|10|7|0|8|3|9|10|0|11|9|12|0|13|9|14|8|3|15|16|15|17|15|18|0|19|20|0|6|0|"
+      end
```
Comment (kcampos) [inline lib/ks/curriculum/curriculum.rb:1584]: Please add a blank line after the "end" so you can clearly see where that if/else block ends

**Pull request merged**